components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/servlet/SAMLArtifactResolveServlet.java

```diff
@@ -0,0 +1,245 @@
+     * handle artifact method.
+     *
+     * @param req  HttpServletRequest object received.
+     * @param resp HttpServletResponse object to be sent.
+     * @throws ServletException
+     * @throws IOException
+     */
+    private void handleRequest(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+
+        try {
+            String id = null;
+            String samlArtifact = null;
+            String issueInstant = null;
+            try {
+                MessageFactory messageFactory = MessageFactory.newInstance();
+                InputStream inStream = req.getInputStream();
+                SOAPMessage soapMessage = messageFactory.createMessage(new MimeHeaders(), inStream);
+                SOAPBody soapBody = soapMessage.getSOAPBody();
+                Iterator iterator = soapBody.getChildElements();
+                while (iterator.hasNext()) {
+                    SOAPBodyElement artifactResolveElement = (SOAPBodyElement) iterator.next();
+                    id = artifactResolveElement.getAttribute("ID");
+                    issueInstant = artifactResolveElement.getAttribute("IssueInstant");
+                    SOAPBodyElement issuerElement = (SOAPBodyElement) artifactResolveElement.getFirstChild();
+                    SOAPBodyElement artifactElement = (SOAPBodyElement) issuerElement.getNextSibling();
+                    samlArtifact = artifactElement.getFirstChild().getNodeValue();
+                    // TODO: 7/4/18 check whether we can convert above to ArtifactResolve object directly
+                }
+            } catch (SOAPException e) {
+                log.error("Invalid SAML Artifact Resolve request received.");
+            }
+
+            if (samlArtifact != null) {
+                handleArtifact(req, resp, samlArtifact, id, issueInstant);
+            } else {
+                log.error("Invalid SAML Artifact Resolve request received.");
+            }
+
+        } finally {
+            SAMLSSOUtil.removeSaaSApplicationThreaLocal();
+            SAMLSSOUtil.removeUserTenantDomainThreaLocal();
+            SAMLSSOUtil.removeTenantDomainFromThreadLocal();
+        }
+    }
+
+    /**
+     * Resolve the received SAML artifact.
+     *
+     * @param req         HttpServletRequest.
+     * @param resp        HttpServletResponse.
+     * @param samlArt     Received SAMl artifact.
+     * @throws IOException
+     * @throws ServletException
+     */
+    private void handleArtifact(HttpServletRequest req, HttpServletResponse resp, String samlArt, String id,
+                                String issueInstant) throws IOException, ServletException {
+
+        if (log.isDebugEnabled()) {
+            log.debug("Resolving SAML artifact : " + samlArt);
+        }
+
+        try {
+
+            Response response = new SAMLSSOArtifactResolver().resolveArtifact(samlArt);
+
```

Comment:
Check for unnecessary new lines in all the places and remove

Reply:
Done.